### PR TITLE
enable linters

### PR DESCRIPTION
**Pull request by vrothberg**
Enable a number of linters and start fixing issues of others. I also added comments to explain _why_ certain linters are disabled.

A healthy exercise since some issues looked serious to me.

@containers/podman-maintainers PTAL

### CODE REVIEW

File: cmd/podman/images/scp.go [hunk 1]
```diff
@@ -264,10 +264,12 @@ func saveToRemote(image, localFile string, tag string, uri *urlP.URL, iden strin
 	run := podman + " image save " + image + " --format=oci-archive --output=" + remoteFile // run ssh image load of the file copied via scp. Files are reverse in this case...
 	_, err = connection.ExecRemoteCommand(dial, run)
 	if err != nil {
-		return nil
+		return err
```
[the remark this answers is not in the document]
Comment (vrothberg) [reply cmd/podman/images/scp.go:267]: @cdoern PTAL

Once enabled, the linter will hopefully force us to drop a comment why we ignore certain errors.

File: libpod/container_internal_linux.go [hunk 2]
```diff
@@ -2740,7 +2740,8 @@ func (c *Container) generateUserPasswdEntry(addedUID int) (string, int, int, err
 	// If a non numeric User, then don't generate passwd
 	uid, err := strconv.ParseUint(userspec, 10, 32)
 	if err != nil {
-		return "", 0, 0, nil
+		// FIXME: this error should probably be returned
+		return "", 0, 0, nil // nolint: nilerr
```
[the remark this answers is not in the document]
Comment (vrothberg) [reply libpod/container_internal_linux.go:2744]: Thanks, I'll add a comment.

File: cmd/podman/machine/init.go [hunk 3]
```diff
@@ -83,7 +83,7 @@ func init() {
 		"reexec", false,
 		"process was rexeced",
 	)
-	flags.MarkHidden("reexec")
+	flags.MarkHidden("reexec") // nolint: errcheck
```
Comment (mheon) [inline cmd/podman/machine/init.go:86]: Would it be better to do `_ := flags.MarkHidden(...)` instead? Seems a bit more clear

Comment (Luap99) [reply cmd/podman/machine/init.go:86]: agree, please do the same for all other places where this is used.

File: pkg/machine/fedora.go [hunk 4]
```diff
@@ -59,7 +59,10 @@ func (f FedoraDownload) Get() *Download {
 func (f FedoraDownload) HasUsableCache() (bool, error) {
 	info, err := os.Stat(f.LocalPath)
 	if err != nil {
-		return false, nil
+		if os.IsNotExist(err) {
```
Comment (Luap99) [inline pkg/machine/fedora.go:62]: use `errors.Is(err, os.ErrNotExist) `

File: libpod/boltdb_state.go [hunk 5]
```diff
@@ -366,7 +366,8 @@ func (s *BoltState) GetDBConfig() (*DBConfig, error) {
 	err = db.View(func(tx *bolt.Tx) error {
 		configBucket, err := getRuntimeConfigBucket(tx)
 		if err != nil {
-			return nil
+			// FIXME: this error should probably be returned
+			return nil // nolint: nilerr
```
[the remark this answers is not in the document]
Comment (mheon) [reply libpod/boltdb_state.go:370]: Concur, this should return an error

File: pkg/machine/fedora.go (hunk 4 above)
Comment (vrothberg) [reply pkg/machine/fedora.go:62]: I can change but I don't really see why.

Comment (Luap99) [reply pkg/machine/fedora.go:62]: Look at the godoc comment: https://pkg.go.dev/os#IsNotExist

File: pkg/specgen/generate/ports_bench_test.go [hunk 6]
```diff
@@ -5,11 +5,13 @@ import (
...
 
 	"github.com/containers/common/libnetwork/types"
+	"github.com/stretchr/testify/require"
 )
 
 func benchmarkParsePortMapping(b *testing.B, ports []types.PortMapping) {
 	for n := 0; n < b.N; n++ {
-		ParsePortMapping(ports, nil)
+		_, err := ParsePortMapping(ports, nil)
+		require.NoError(b, err)
```
Comment (Luap99) [inline pkg/specgen/generate/ports_bench_test.go:14]: This is a benchmark, I am not sure If we want to slow it down with an extra unrelated function call.

File: pkg/checkpoint/crutils/checkpoint_restore_utils.go [hunk 7]
```diff
@@ -99,13 +99,12 @@ func CRRemoveDeletedFiles(id, baseDirectory, containerRootDirectory string) erro
...
 	rootfsDiffPath := filepath.Join(baseDirectory, metadata.RootFsDiffTar)
-	if _, err := os.Stat(rootfsDiffPath); err != nil {
-		// Only do this if a rootfs-diff.tar actually exists
-		return nil
-	}
-
+	// Only do this if a rootfs-diff.tar actually exists
 	rootfsDiffFile, err := os.Open(rootfsDiffPath)
 	if err != nil {
+		if os.IsNotExist(err) {
```
Comment (Luap99) [inline pkg/checkpoint/crutils/checkpoint_restore_utils.go:105]: errros.Is

File: libpod/container_api.go [hunk 8]
```diff
@@ -921,7 +921,11 @@ func (c *Container) Stat(ctx context.Context, containerPath string) (*define.Fil
 		if err != nil {
 			return nil, err
 		}
-		defer c.unmount(false)
+		defer func() {
+			if err := c.unmount(false); err != nil {
+				logrus.Errorf("Unmounting container %s: %v", c.ID(), err)
```
Comment (rhatdan) [inline libpod/container_api.go:926]: Should we check if the container is already unmounted?  And only write at Info level in that case.   I am a little worried about a race condition, between --rm and podman container cleanup.

File: cmd/rootlessport/main.go [hunk 9]
```diff
@@ -253,9 +253,9 @@ func serve(listener net.Listener, pm rkport.Manager) {
 		ctx := context.TODO()
 		err = handler(ctx, conn, pm)
 		if err != nil {
-			conn.Write([]byte(err.Error()))
+			conn.Write([]byte(err.Error())) // nolint: errcheck
```
Comment (Luap99) [inline cmd/rootlessport/main.go:256]: use `_ = conn.Write...` instead of disabling the linter, same below

File: cmd/podman/validate/args.go [hunk 10]
```diff
@@ -27,7 +27,7 @@ func SubCommandExists(cmd *cobra.Command, args []string) error {
 		}
 		return errors.Errorf("unrecognized command `%[1]s %[2]s`\n\nDid you mean this?\n\t%[3]s\n\nTry '%[1]s --help' for more information.", cmd.CommandPath(), args[0], strings.Join(suggestions, "\n\t"))
 	}
-	cmd.Help()
+	cmd.Help() // nolint: errcheck
```
Comment (Luap99) [inline cmd/podman/validate/args.go:30]: same here use `_ = ...`

File: libpod/events/logfile.go [hunk 11]
```diff
@@ -59,7 +60,9 @@ func (e EventLogFile) Read(ctx context.Context, options ReadOptions) error {
 		}
 		go func() {
 			time.Sleep(time.Until(untilTime))
-			t.Stop()
+			if err := t.Stop(); err != nil {
+				logrus.Errorf("Stopping logger: %v", err)
```
Comment (rhatdan) [inline libpod/events/logfile.go:64]: Is this a potential race condition where the container is already stopped and/or deleted?

File: pkg/rootless/rootless.go [hunk 12]
```diff
@@ -16,12 +16,15 @@ import (
 // file.
 func TryJoinPauseProcess(pausePidPath string) (bool, int, error) {
 	if _, err := os.Stat(pausePidPath); err != nil {
-		return false, -1, nil
+		if os.IsNotExist(err) {
```
Comment (Luap99) [inline pkg/rootless/rootless.go:19]: errors.Is

File: libpod/events/logfile.go (hunk 11 above)
Comment (vrothberg) [reply libpod/events/logfile.go:64]: I don't think so. `t` is the `Tail` to read the logs, not the container.

**Review by Luap99 — COMMENTED** [review]
LGTM

**Comment by vrothberg** [comment]
Ready to merge

**Pull request merged**